pyHS100/antitheft.py

```diff
@@ -0,0 +1,199 @@
+"""
+pyHS100
+Python library supporting TP-Link Smart Plugs/Switches (HS100/HS110/Hs200).
+
+The communication protocol was reverse engineered by Lubomir Stroetmann and
+Tobias Esser in 'Reverse Engineering the TP-Link HS110':
+https://www.softscheck.com/en/reverse-engineering-tp-link-hs110/
+
+This library reuses codes and concepts of the TP-Link WiFi SmartPlug Client
+at https://github.com/softScheck/tplink-smartplug, developed by Lubomir
+Stroetmann which is licensed under the Apache License, Version 2.0.
+
+You may obtain a copy of the license at
+http://www.apache.org/licenses/LICENSE-2.0
+"""
+
+
+class AntiTheft(object):
+    def __init__(self, _device, _query_helper):
+        self._device = _device
+        self._query_helper = _query_helper
+
+    @property
+    def _get_antitheft(self):
+        """
+        Gets all rules.
+
+        :return: list rules
+        :rtype: list
+        :raises SmartPlugException: on error
+        """
+
+        return self._query_helper(
+            "anti_theft",
+            "get_rules"
+        )
+
+    def __iter__(self):
+        """
+        Iterates all rules.
+
+        :return: iterator (name, rule)
+        :rtype: iterator
+        :raises SmartPlugException: on error
+        """
+
+        for antitheft in self._get_antitheft:
+            yield antitheft['name'], antitheft
+
+    def __contains__(self, name):
+        """
+        Checks if rule name exists.
+
+        :param str name: name of the rule
+        :return: bool True if name exists else False
+        :rtype: bool
+        :raises SmartPlugException: on error
+        """
+
+        for n, _ in self:
+            if n == name:
+                return True
+        return False
+
+    def __setitem__(self, name, rule):
+        """
+        Edits or Adds a new rule.
+
+        If a rule exists with the same name it will be updated.
+        Otherwise it will be added.
+
+        Value must be a dict with the following key, value pairs
+        "stime_opt":int(0)
+        "wday":list(0, 1, 0, 1, 1, 0, 0), 0 or a 1 for each day of the week
+                                          1 representing a scheduled day
+        "smin":int(1014)
+        "enable":int(1), 0 or a 1, 1 being enabled
+        "frequency": int(5)
+        "repeat": int(1), 0 or a 1, 1 to repeat the schedule
+        "etime_opt": int(-1)
+        "duration": int(2)
+        "lastfor": int(1)
+        "month": int(0), 0 for all months
+        "year": int(0), 0 for all years
+        "longitude": int/float(0), 0 for all locations
+        "day": int(0), 0 for all days of the month
+        "force": int(0), 0 or 1, 1 to force schedule
+        "latitude": int/float(0), 0 for all locations
+        "emin": int(0)
+
```

Comment:
Would it make sense to have a wrapper class for this information? In any case I think it would make sense to use e.g. voluptuous (https://github.com/yadutaf/voluptuous) to enforce correct input values.